### PR TITLE
JAMES-2263 Document mailet error handling

### PR DESCRIPTION
**Pull request by chibenwa**
Also ship some additional links and corrections to ease documentation reading.

### CODE REVIEW

File: src/site/xdoc/server/config-mailetcontainer.xml [hunk 1]
```diff
@@ -77,6 +77,28 @@
       pair of ("processor", "spam").</p>
     
     </subsection>
+
+    <subsection name="Error handling">
+      <p>If an exception is encountered during the execution of a mailet or a matcher, the default behaviour is to
+      process the mail using the <strong>error</strong> processor.</p>
+
+      <p>The <strong>onMailetException</strong> property allow you to override this behaviour. You can specify an other
```
Comment (rouazana) [inline src/site/xdoc/server/config-mailetcontainer.xml]: allows

File: src/site/xdoc/server/config-mailetcontainer.xml [hunk 2]
```diff
@@ -77,6 +77,28 @@
...
+      processor than the error one for handling the errors of this mailet. The <strong>ignore</strong> special value also
+      allow to continue processing and ignore the error.</p>
+
+      <p>Moreover, the <strong>onMatcherException</strong> allow you to override matcher error handling logic. You can
```
Comment (rouazana) [inline src/site/xdoc/server/config-mailetcontainer.xml]: allows

File: src/site/xdoc/server/config-mailetcontainer.xml [hunk 3]
```diff
@@ -77,6 +77,28 @@
...
+          specify an other processor than the error one for handling the errors of this mailet. The <strong>matchall</strong>
+          special value also allow you to match all recipients when there is an error. The <strong>nomatch</strong>
```
Comment (rouazana) [inline src/site/xdoc/server/config-mailetcontainer.xml]: allows

File: src/site/xdoc/server/config-mailetcontainer.xml [hunk 4]
```diff
@@ -77,6 +77,28 @@
...
+          special value also allow you to match no recipients when there is an error.</p>
```
Comment (rouazana) [inline src/site/xdoc/server/config-mailetcontainer.xml]: allows

File: src/site/xdoc/server/feature-mailetcontainer.xml [hunk 5]
```diff
@@ -47,16 +47,18 @@
...
+      a yes or no issue.  Instead, the match method returns a collection of matched recipients.  If the
       this collection of matched recipients is empty, the mailet is not invoked.  If the collection of
-      matched recipients is the entire set of original recipients, the mail is then processed by the 
+      matched recipients is the entire set of original recipients, the mail is then processed by the
       associated mailet.  Finally, if the matcher only matches a proper subset of the original recipients,
       the original mail is duplicated.  The recipients for one message are set to the matched recipients,
       and that message is processed by the mailet.  The recipients for the other mail are set to the
       non-matching recipients, and that message is not processed by the mailet.</p>
+
+      <p>Note that you can specify per mailet and per matcher detailed error handling logic.</p>
```
Comment (rouazana) [inline src/site/xdoc/server/feature-mailetcontainer.xml]: no link?

File: src/site/xdoc/server/config-mailetcontainer.xml (hunk 1 above)
Comment (mbaechler) [inline src/site/xdoc/server/config-mailetcontainer.xml]: s/an other/another

File: src/site/xdoc/server/config-mailetcontainer.xml [hunk 6]
```diff
@@ -77,6 +77,28 @@
       pair of ("processor", "spam").</p>
     
     </subsection>
+
+    <subsection name="Error handling">
+      <p>If an exception is encountered during the execution of a mailet or a matcher, the default behaviour is to
+      process the mail using the <strong>error</strong> processor.</p>
+
+      <p>The <strong>onMailetException</strong> property allow you to override this behaviour. You can specify an other
+      processor than the error one for handling the errors of this mailet. The <strong>ignore</strong> special value also
```
Comment (mbaechler) [inline src/site/xdoc/server/config-mailetcontainer.xml]: please escape by any mean words that are identifiers (like `error` here)

File: src/site/xdoc/server/config-mailetcontainer.xml (hunk 2 above)
Comment (mbaechler) [inline src/site/xdoc/server/config-mailetcontainer.xml]: you can remove "logic" from the sentence, I think it makes things easier to read and we don't loose much

File: src/site/xdoc/server/config-mailetcontainer.xml [hunk 7]
```diff
@@ -77,6 +77,28 @@
...
+      allow to continue processing and ignore the error.</p>
+
+      <p>Moreover, the <strong>onMatcherException</strong> allow you to override matcher error handling logic. You can
+          specify an other processor than the error one for handling the errors of this mailet. The <strong>matchall</strong>
```
Comment (mbaechler) [inline src/site/xdoc/server/config-mailetcontainer.xml]: another

File: src/site/xdoc/server/feature-mailetcontainer.xml (hunk 5 above)
Comment (mbaechler) [inline src/site/xdoc/server/feature-mailetcontainer.xml]: why "detailed" ? do you mean specific ?

Comment (chibenwa) [reply src/site/xdoc/server/feature-mailetcontainer.xml]: That differs only from an ancor from a link specified above. Anyway, added.

Comment (chibenwa) [reply src/site/xdoc/server/feature-mailetcontainer.xml]: Removed

**Pull request merged**